### PR TITLE
update numpy and fix numpy records

### PR DESCRIPTION
**Pull request by feixie**
Thanks for contributing! Please remove any top-level sections that do not apply to your changes.

- [x] `make format && make documentation` has been run.

# New variable

- [ ] Label field added
- [ ] Documentation field added
- [ ] Unit field added
- [ ] Default value field added if relevant
- [ ] Variable name follows conventions
- [ ] Unit test(s) added
- [ ] Integration test(s) added if relevant
- [x] Issues this PR fixes linked

## What's changed

Bumped numpy version to match what's in openfisca-core. Bring in changes from openfisca-core to fix `TypeError: ufunc 'isnan' not supported for the input types`

# Bug fix

- [ ] Regression test added
- [x] Regression test passing

## What this fixes and how it's fixed

Fixes this issue: https://github.com/PolicyEngine/policyengine-us/issues/2802

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Could you try `make format`, and populate `changelog_entry.yaml` instead of `changelog.yaml`?

**Comment by feixie** [comment]
> Could you try `make format`, and populate `changelog_entry.yaml` instead of `changelog.yaml`?

Fixed

**Review by MaxGhenis — APPROVED** [review]
Thanks Fei! LGTM, just requesting a review from Nikhil.

File: policyengine_core/parameters/vectorial_parameter_node_at_instant.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def build_from_node(
         node: "ParameterNode",
     ) -> "VectorialParameterNodeAtInstant":
         VectorialParameterNodeAtInstant.check_node_vectorisable(node)
-        subnodes_name = node._children.keys()
+        subnodes_name = sorted(node._children.keys())
```
Comment (nikhilwoodruff) [inline policyengine_core/parameters/vectorial_parameter_node_at_instant.py:26]: Sorry just remind me why this is necessary? Thanks

**Review by nikhilwoodruff — APPROVED** [review]
OK to merge since the numpy update is obviously a massive plus- just curious about the `sorted` usage but I doubt it adds that big a time cost. Thanks @feixie!

File: policyengine_core/parameters/vectorial_parameter_node_at_instant.py (hunk 1 above)
Comment (feixie) [reply policyengine_core/parameters/vectorial_parameter_node_at_instant.py:26]: I got this change from this commit: https://github.com/openfisca/openfisca-core/commit/c0782fa5e98bb5aab695863eed38c2856366fcc8
Without it I get the TypeError on unit test runs.

**Pull request merged**